vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1212,6 +1234,89 @@ def _gather_mm_embeddings(
                 mm_embeds.append(mm_embeds_item)
         return mm_embeds
 
+    def _extract_encoder_inputs(
+        self,
+        scheduler_output: "SchedulerOutput",
+    ) -> dict[str, torch.Tensor]:
```

Comment:
I think this method and the one below would best fit within the `_prepare_inputs` context, so that we could think about re-using a singly allocated on-device tensor for both encoder_inputs.. and input_features like it happens for the rest of params in V1.

We can also leave this as future optimization work.

Reply:
Thanks for the feedback. Let me know if you think we need to block on it.